Source/Array.swift

```diff
@@ -40,6 +40,11 @@ public func >>-<T, U>(a: [T], f: T -> [U]) -> [U] {
     return a.flatMap(f)
 }
 
+@availability(*, unavailable, message="function (T -> U) is not equal to function (T -> [U])")
+public func >>-<T, U>(a: [T], f: T -> U) -> [U] {
+    return a.map(f)
+}
```

Comment:
No, it doesn't. That would be horrible. But without this, it throws the Optional version of the warning. The Array warning serves to override that one and provide the correct error.